### ISSUE TITLE
"_computeOfferPrice" miscalculation

### ISSUE THREAD

**Issue by ghzwrapper** [dapperlabs/cryptokitties-bounty-2#1]
### Description
The calculation of `_computeOfferPrice` defies the normal definition of a percentage fee. 

Instead of: 
`return _total * (1e4 - _offerCut)/ 1e4;`

the current  formula is:
`return _total * 1e4 / (1e4 + _offerCut);`

### Scenario
Computing the offer (in almost all* cases) yields a slightly incorrect figure for the computed offer price.

#### For example:
If I were to offer **1 ETH** for a CryptoKitty, and the contract defined an `_offerCut` of **200** basis points (**2%**), I would expect my offer after fees to be **0.98 ETH**. 

However, according to the current formula, my offer after fees would be **0.980392157 ETH**.

*For some smaller bids, it's possible that integer math would yield equivalent figures for the currently computed offer price and the expected computed offer price.

### Impact
The contract owners would be receiving an offerCut that is slightly less than what is expected, and the offer recipient will be receiving an offer that is slightly more than what is expected.

Nothing is being misrepresented for the end user if the contract logic is followed, but the contract logic does not define the expected calculation of a percentage cut.

### Fix
Replace the current implementation of `_computeOfferPrice` with:
`return _total * (1e4 - _offerCut) / 1e4;`

**Comment by michaelKim4736** [comment]
ghzwrapper correctly points out this issue. However, his suggested fix also returns an incorrect result. He suggested:

return _total * (1e4 - _offerCut / 1e4)

But the correct formula is:

return _total * (1e4 - _offerCut) / 1e4

due to order of operations.

**Comment by ghzwrapper** [comment]
Thanks @michaelKim4736 - corrected it in the description

**Comment by zhangchiqing** [comment]
Hi @ghzwrapper

Thanks for the feedback. I agree our usage of the percentage fee is a bit different than the normal one. The reason we decided to use the current formula is to ensure the owner sees an offer price that the bidder would like to offer. 

For example, let's say the `_offerCut ` is 200 basis points (2%), if a bidder would like to offer **1ETH** to the owner to buy a kitty, then the **1ETH** is the offer price, the total value would be calculated on the client, which is on our CrypoKitties website, to be **1.02 ETH** (1 * 1.02). Once the contract receives a `createOffer` call with **1.02 ETH** in the value, it will calculate the offer price (P) to be **1ETH** (1.02 * 10000 / (10000 + 200)), which is the original offer price that the bidder would like to offer to the owner. So there is no math issue, the calculation works as expected.

Closing now. Thanks for the feedback.
